tests/flight/client.rs

```diff
@@ -0,0 +1,65 @@
+use crate::flight::*;
+use crate::statements::create_table_and_some_partitions;
+
+#[tokio::test]
+async fn test_basic_queries() -> Result<()> {
+    let (context, addr, flight) = start_flight_server().await;
+    create_table_and_some_partitions(context.as_ref(), "flight_table", None).await;
+    tokio::task::spawn(flight);
+
+    // Create the channel and the client
+    let port = addr.port();
+    let channel = Channel::from_shared(format!("http://localhost:{port}"))
+        .expect("Endpoint created")
+        .connect()
+        .await
+        .expect("Channel connected");
+
+    let mut client = FlightClient::new(channel);
+
+    // Test the handshake works
+    let _ = client.handshake("test").await.expect("error handshaking");
+
+    // Try out the actual query
+    let cmd = CommandStatementQuery {
+        query: "SELECT * FROM flight_table".to_string(),
```

Comment:
I'd test some basic DDL here (CREATE TABLE / CREATE EXTERNAL TABLE maybe) and a slightly more complex SELECT. Maybe also test two queries getting executed at the same time, with interleaving? i.e. start Q1, start Q2, get results for Q2, get results for Q1.